File: clutch-partners-android/app/src/main/java/com/clutch/partners/ui/screens/auth/PartnerTypeSelectorScreen.kt
package com.clutch.partners.ui.screens.auth

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.navigation.NavController
import com.clutch.partners.ClutchPartnersTheme
import com.clutch.partners.R
import com.clutch.partners.data.model.PartnerType
import com.clutch.partners.navigation.Screen
import com.clutch.partners.utils.LanguageManager
import com.clutch.partners.viewmodel.AuthViewModel

@Composable
fun PartnerTypeSelectorScreen(
    navController: NavController,
    viewModel: AuthViewModel = hiltViewModel()
) {
    val context = LocalContext.current
    var selectedType by remember { mutableStateOf<PartnerType?>(null) }
    val currentLanguage = LanguageManager.getCurrentLanguage(context)
    
    ClutchPartnersTheme {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(MaterialTheme.colorScheme.background)
        ) {
            // Header with logo and title
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Partners Logo (White)
                Image(
                    painter = painterResource(id = R.drawable.partners_logo_white),
                    contentDescription = "Partners Logo",
                    modifier = Modifier.size(60.dp)
                )
                
                Spacer(modifier = Modifier.height(16.dp))
                
                // Title
                Text(
                    text = if (currentLanguage == "ar") "اختر نوع عملك" else "Select Your Business Type",
                    style = MaterialTheme.typography.headlineMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onBackground,
                    textAlign = TextAlign.Center
                )
                
                Spacer(modifier = Modifier.height(8.dp))
                
                Text(
                    text = if (currentLanguage == "ar") "اختر النوع الذي يصف عملك بشكل أفضل" else "Choose the type that best describes your business",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.7f),
                    textAlign = TextAlign.Center
                )
            }
            
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(horizontal = 24.dp)
            ) {
                Spacer(modifier = Modifier.height(16.dp))
                
                LazyVerticalGrid(
                    columns = GridCells.Fixed(2),
                    horizontalArrangement = Arrangement.spacedBy(16.dp),
                    verticalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    items(PartnerType.values().toList()) { type ->
                        PartnerTypeCard(
                            type = type,
                            isSelected = selectedType == type,
                            onClick = { selectedType = type },
                            currentLanguage = currentLanguage
                        )
                    }
                }
                
                Spacer(modifier = Modifier.weight(1f))
                
                Button(
                    onClick = { 
                        selectedType?.let {
                            navController.navigate(Screen.Auth.route)
                        }
                    },
                    enabled = selectedType != null,
                    modifier = Modifier.fillMaxWidth(),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = if (selectedType != null) Color(0xFF242424) else MaterialTheme.colorScheme.primary.copy(alpha = 0.3f),
                        contentColor = if (selectedType != null) Color.White else MaterialTheme.colorScheme.onPrimary.copy(alpha = 0.5f)
                    )
                ) {
                    Text(
                        text = if (currentLanguage == "ar") "متابعة" else "Continue",
                        fontWeight = FontWeight.SemiBold
                    )
                }
            }
        }
    }
}

@Composable
fun PartnerTypeCard(
    type: PartnerType,
    isSelected: Boolean,
    onClick: () -> Unit,
    currentLanguage: String
) {
    val (title, description, icon) = when (type) {
        PartnerType.REPAIR_CENTER -> if (currentLanguage == "ar") Triple(
            "مركز إصلاح",
            "خدمات إصلاح وصيانة السيارات",
            Icons.Default.Build
        ) else Triple(
            "Repair Center",
            "Auto repair and maintenance services",
            Icons.Default.Build
        )
        PartnerType.AUTO_PARTS -> if (currentLanguage == "ar") Triple(
            "قطع غيار",
            "بيع قطع غيار وإكسسوارات السيارات",
            Icons.Default.Settings
        ) else Triple(
            "Auto Parts",
            "Sell automotive parts and accessories",
            Icons.Default.Settings
        )
        PartnerType.ACCESSORIES -> if (currentLanguage == "ar") Triple(
            "إكسسوارات",
            "إكسسوارات وتخصيص السيارات",
            Icons.Default.Star
        ) else Triple(
            "Accessories",
            "Car accessories and customization",
            Icons.Default.Star
        )
        PartnerType.IMPORTER -> if (currentLanguage == "ar") Triple(
            "مستورد",
            "استيراد وتوزيع منتجات السيارات",
            Icons.Default.LocalShipping
        ) else Triple(
            "Importer",
            "Import and distribute automotive products",
            Icons.Default.LocalShipping
        )
        PartnerType.MANUFACTURER -> if (currentLanguage == "ar") Triple(
            "مصنع",
            "تصنيع منتجات السيارات",
            Icons.Default.Settings
        ) else Triple(
            "Manufacturer",
            "Manufacture automotive products",
            Icons.Default.Settings
        )
        PartnerType.SERVICE_CENTER -> if (currentLanguage == "ar") Triple(
            "مركز خدمة",
            "خدمات شاملة للسيارات",
            Icons.Default.AutoFixHigh
        ) else Triple(
            "Service Center",
            "Comprehensive automotive services",
            Icons.Default.AutoFixHigh
        )
    }
    
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clickable { onClick() },
        colors = CardDefaults.cardColors(
            containerColor = if (isSelected) Color.White else MaterialTheme.colorScheme.surface
        ),
        border = if (isSelected) 
            CardDefaults.outlinedCardBorder().copy(
                brush = androidx.compose.ui.graphics.SolidColor(MaterialTheme.colorScheme.primary)
            )
        else null
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                modifier = Modifier.size(32.dp),
                tint = if (isSelected) Color.Gray else MaterialTheme.colorScheme.onSurface
            )
            
            Spacer(modifier = Modifier.height(8.dp))
            
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center,
                color = if (isSelected) Color.Gray else MaterialTheme.colorScheme.onSurface
            )
            
            Spacer(modifier = Modifier.height(4.dp))
            
            Text(
                text = description,
                style = MaterialTheme.typography.bodySmall,
                textAlign = TextAlign.Center,
                color = if (isSelected) Color.Gray.copy(alpha = 0.7f) else MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
            )
        }
    }
}

File: clutch-partners-ios/ClutchPartners/Views/Auth/PartnerTypeSelectorScreen.swift
import SwiftUI

struct PartnerTypeSelectorScreen: View {
    @EnvironmentObject var router: AppRouter
    @State private var selectedType: PartnerType?
    
    private let partnerTypes = [
        PartnerType(id: "repair", title: "Repair Center", description: "Auto repair and maintenance services", icon: "wrench.and.screwdriver.fill", color: .blue),
        PartnerType(id: "parts", title: "Auto Parts", description: "Parts and accessories retail", icon: "car.fill", color: .green),
        PartnerType(id: "accessories", title: "Accessories", description: "Car accessories and modifications", icon: "star.fill", color: .orange),
        PartnerType(id: "importer", title: "Importer", description: "Import and distribute automotive products", icon: "shippingbox.fill", color: .purple),
        PartnerType(id: "manufacturer", title: "Manufacturer", description: "Manufacture automotive components", icon: "gearshape.fill", color: .red),
        PartnerType(id: "service", title: "Service Center", description: "General automotive services", icon: "wrench.fill", color: .teal)
    ]
    
    var body: some View {
        VStack(spacing: 30) {
            VStack(spacing: 10) {
                Text("Choose Your Business Type")
                    .font(.largeTitle)
                    .fontWeight(.bold)
                
                Text("Select the type of automotive business you operate")
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                    .multilineTextAlignment(.center)
            }
            .padding(.top, 50)
            
            LazyVGrid(columns: [
                GridItem(.flexible()),
                GridItem(.flexible())
            ], spacing: 20) {
                ForEach(partnerTypes, id: \.id) { type in
                    PartnerTypeCard(
                        type: type,
                        isSelected: selectedType?.id == type.id
                    ) {
                        selectedType = type
                    }
                }
            }
            .padding(.horizontal)
            
            Spacer()
            
            Button(action: {
                if let selected = selectedType {
                    router.navigateTo(.auth)
                }
            }) {
                Text("Continue")
                    .font(.headline)
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .padding()
                    .background(selectedType != nil ? Color.blue : Color.gray)
                    .cornerRadius(10)
            }
            .disabled(selectedType == nil)
            .padding(.horizontal)
            .padding(.bottom, 50)
        }
        .background(Color(.systemBackground))
    }
}

struct PartnerType {
    let id: String
    let title: String
    let description: String
    let icon: String
    let color: Color
}

struct PartnerTypeCard: View {
    let type: PartnerType
    let isSelected: Bool
    let onTap: () -> Void
    
    var body: some View {
        Button(action: onTap) {
            VStack(spacing: 15) {
                Image(systemName: type.icon)
                    .font(.system(size: 40))
                    .foregroundColor(isSelected ? .white : type.color)
                
                VStack(spacing: 5) {
                    Text(type.title)
                        .font(.headline)
                        .fontWeight(.semibold)
                        .foregroundColor(isSelected ? .white : .primary)
                    
                    Text(type.description)
                        .font(.caption)
                        .foregroundColor(isSelected ? .white.opacity(0.8) : .secondary)
                        .multilineTextAlignment(.center)
                }
            }
            .frame(maxWidth: .infinity)
            .padding()
            .background(isSelected ? type.color : Color(.systemGray6))
            .cornerRadius(15)
            .overlay(
                RoundedRectangle(cornerRadius: 15)
                    .stroke(isSelected ? type.color : Color.clear, lineWidth: 2)
            )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

#Preview {
    PartnerTypeSelectorScreen()
        .environmentObject(AppRouter())
}
